workspaces/util/src/chunkBySize.ts

```diff
@@ -2,7 +2,23 @@ export function* chunkBySize<T>(
   array: readonly T[],
   chunkSize: number,
 ): Generator<T[], void, void> {
+  if (
+    typeof chunkSize !== "number" ||
+    chunkSize < 1 ||
+    !Number.isInteger(chunkSize)
+  ) {
+    throw new RangeError("Chunk size must be a positive integer!");
+  }
+
   for (let i = 0; i < array.length; i += chunkSize) {
-    yield array.slice(i, i + chunkSize);
+    const chunk = array.slice(i, i + chunkSize);
+
+    for (const item of chunk) {
+      if (item === undefined) {
+        throw new TypeError("Array elements cannot be undefined!");
+      }
+    }
```

Comment:
Apologies if it wasn't clear! I don't think we need to defend against this -- it's reasonable for an array to contain undefined elements. It might just be nice to have a test that includes it among the data. For example if we `chunkBySize([null, undefined, false, true, new Set(), 5, "hello"], 3)` we'd expect to see `[null, undefined, false]`, `[true, new Set(), 5]`, `["hello"]`.